### PR TITLE
Re-vendor to pick up client-go@kubernetes-1.15.1

### PR DESCRIPTION
**Pull request by MrHohn**
This is similar to https://github.com/kubernetes-incubator/cluster-proportional-vertical-autoscaler/pull/20.

I ended up need to work around the scale function a bit because the generated clients for Scale types were dropped (https://github.com/kubernetes/kubernetes/pull/70550).

cc @mikedanese

### CODE REVIEW

**Comment by MrHohn** [comment]
The previous client-go was cut from 1.6 Kubernetes, which is pretty old.

File: pkg/autoscaler/k8sclient/k8sclient.go [hunk 1]
```diff
@@ -226,16 +230,39 @@ func (k *k8sClient) UpdateReplicas(expReplicas int32) (prevRelicas int32, err er
...
-		_, err = k.clientset.Extensions().Scales(k.target.namespace).Update(k.target.kind, scale)
+		_, err = k.updateScaleExtensionsV1beta1(k.target, scale)
 		if err != nil {
 			return 0, err
 		}
 	}
 	return prevRelicas, nil
 }
 
+func (k *k8sClient) getScaleExtensionsV1beta1(target *scaleTarget) (*extensionsv1beta1.Scale, error) {
```
Comment (MrHohn) [inline pkg/autoscaler/k8sclient/k8sclient.go]: These are the work around I mentioned.

**Comment by mikedanese** [comment]
client-go version in this PR (commit 8e956561bbf5) contains the fixes that we need.

**Comment by MrHohn** [comment]
@mikedanese Thanks for taking the look, adding @bowei to keep this rolling.
/assign @bowei

**Pull request merged**